### PR TITLE
Add docs for combine patch types

### LINKED ISSUE #2362: Document Combine{From,To}Composite patch type
### What problem are you facing?

Immediately after I merged https://github.com/crossplane/crossplane/pull/2352 I realised we didn't update the documentation to highlight the new patch type and how to use it.

### How could Crossplane help solve your problem?

We could update the documentation to highlight the new patch type and how to use it. :)

### PR DESCRIPTION
**Pull request by benagricola**
### Description of your changes
Adds documentation for the recently merged `Combine` patch types.

Fixes #2362

Signed-off-by: Ben Agricola <bagricola@squiz.co.uk>

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.

This should solve the immediate need of making sure all of the patch types are documented, but it might be worth exploring the more "advanced" composition techniques in a separate doc (specifically - bidirectional and combine patching).

### How has this code been tested
I have read it a number of times 🤷‍♂️

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: docs/concepts/composition.md [hunk 1]
```diff
@@ -495,6 +527,12 @@ spec:
     - type: ToCompositeFieldPath
       fromFieldPath: "status.atProvider.fullyQualifiedDomainName"
       toFieldPath: "status.address"
+
+    # It is also possible to combine multiple values from the composed resource
```
Comment (muvaf) [inline docs/concepts/composition.md:531]: Can you add an example for that, too?

Comment (benagricola) [reply docs/concepts/composition.md:537]: Added. Was struggling to find a good example, but I think exposing a DSN based on the `administratorLogin` input and the `fullyQualifiedDomainName` status is a topical usage of the combine patch 👍 Thanks for the URL suggestion, I think this is along the same lines.

Comment (muvaf) [reply docs/concepts/composition.md:537]: Yeah that's a great example!

**Review by muvaf — APPROVED** [review]
LGTM! Let me know when it's ready to merge.

**Comment by benagricola** [comment]
> LGTM! Let me know when it's ready to merge.

Just had one final read and I'm happy with it if there's no other concerns :)

**Pull request merged**